### ISSUE TITLE
terminfo not installed?

### ISSUE THREAD

**Issue by bcbnz** [barnumbirr/alacritty-debian#3]
Thanks for packaging! Would it be possible to get the terminfo installed as part of the package?

**Comment by barnumbirr** [comment]
Hey there,
IIRC correctly alacritty terminfo should already be taken care of by the package.
Can you please post the output of the following command:
```bash
infocmp alacritty
```
Please also specify what version of the package you're currently running.

Cheers.

**Comment by bcbnz** [comment]
```shell
# infocmp alacritty
infocmp: couldn't open terminfo file /usr/share/terminfo/a/alacritty
```
This is using `alacritty_0.7.2-2_amd64_debian_buster.deb` from the latest release on a fresh Debian Buster container (I do not have access to the physical machines right now, but I was getting the same message on them until I manually installed the terminfo).

Also the contents of the package do not include the terminfo:
```shell
# dpkg -c alacritty_0.7.2-2_amd64_debian_buster.deb
drwxr-xr-x root/root         0 2021-03-14 01:00 ./
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/bin/
-rwxr-xr-x root/root   5734408 2021-03-14 01:00 ./usr/bin/alacritty
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/share/
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/share/applications/
-rw-r--r-- root/root       333 2021-03-14 01:00 ./usr/share/applications/com.alacritty.Alacritty.desktop
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/share/bash-completion/
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/share/bash-completion/completions/
-rw-r--r-- root/root      1597 2021-03-14 01:00 ./usr/share/bash-completion/completions/alacritty
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/share/doc/
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/share/doc/alacritty/
-rw-r--r-- root/root       411 2021-03-14 01:00 ./usr/share/doc/alacritty/changelog.Debian.gz
-rw-r--r-- root/root     13600 2021-03-14 01:00 ./usr/share/doc/alacritty/changelog.gz
-rw-r--r-- root/root       945 2021-03-14 01:00 ./usr/share/doc/alacritty/copyright
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/share/doc/alacritty/examples/
-rw-r--r-- root/root     28229 2021-03-14 01:00 ./usr/share/doc/alacritty/examples/alacritty.yml
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/share/fish/
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/share/fish/completions/
-rw-r--r-- root/root      1552 2021-03-14 01:00 ./usr/share/fish/completions/alacritty.fish
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/share/man/
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/share/man/man1/
-rw-r--r-- root/root       960 2021-03-14 01:00 ./usr/share/man/man1/alacritty.1.gz
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/share/pixmaps/
-rw-r--r-- root/root     17254 2021-03-14 01:00 ./usr/share/pixmaps/com.alacritty.Alacritty.svg
drwxr-xr-x root/root         0 2021-03-14 01:00 ./usr/share/vendor-completions/
-rw-r--r-- root/root      1012 2021-03-14 01:00 ./usr/share/vendor-completions/_alacritty
```

And manually extracting the package and looking at the postinst script, it does not install it either.

**Comment by bcbnz** [comment]
I'm not very familiar with Debian packaging standards. Would it make sense to have a separate `alacritty-terminfo` package which can be installed on e.g., headless servers where alacritty would not be run, and then have it listed in the recommends field of `alacritty`? If I understand correctly, the default apt behaviour is to also install recommended packages, though I am not sure if that applies to dpkg installations from packages not in a repository.

**Comment by barnumbirr** [comment]
This is weird. Here´s the output on my machine (alacritty `0.7.2-2` on Debian Unstable):

```bash
[user@host:~]$ infocmp alacritty
#	Reconstructed via infocmp from file: /etc/terminfo/a/alacritty
alacritty|alacritty terminal emulator,
	am, bce, ccc, hs, km, mc5i, mir, msgr, npc, xenl,
	colors#0x100, cols#80, it#8, lines#24, pairs#0x7fff,
	acsc=``aaffggiijjkkllmmnnooppqqrrssttuuvvwwxxyyzz{{||}}~~,
	bel=^G, blink=\E[5m, bold=\E[1m, cbt=\E[Z, civis=\E[?25l,
	clear=\E[H\E[2J, cnorm=\E[?12l\E[?25h, cr=\r,
	csr=\E[%i%p1%d;%p2%dr, cub=\E[%p1%dD, cub1=^H,
	cud=\E[%p1%dB, cud1=\n, cuf=\E[%p1%dC, cuf1=\E[C,
	cup=\E[%i%p1%d;%p2%dH, cuu=\E[%p1%dA, cuu1=\E[A,
	cvvis=\E[?12;25h, dch=\E[%p1%dP, dch1=\E[P, dim=\E[2m,
	dl=\E[%p1%dM, dl1=\E[M, dsl=\E]2;\007, ech=\E[%p1%dX,
	ed=\E[J, el=\E[K, el1=\E[1K, flash=\E[?5h$<100/>\E[?5l,
	fsl=^G, home=\E[H, hpa=\E[%i%p1%dG, ht=^I, hts=\EH,
	ich=\E[%p1%d@, il=\E[%p1%dL, il1=\E[L, ind=\n,
	indn=\E[%p1%dS,
	initc=\E]4;%p1%d;rgb\:%p2%{255}%*%{1000}%/%2.2X/%p3%{255}%*%{1000}%/%2.2X/%p4%{255}%*%{1000}%/%2.2X\E\\,
	invis=\E[8m, is2=\E[!p\E[?3;4l\E[4l\E>, kDC=\E[3;2~,
	kEND=\E[1;2F, kHOM=\E[1;2H, kIC=\E[2;2~, kLFT=\E[1;2D,
	kNXT=\E[6;2~, kPRV=\E[5;2~, kRIT=\E[1;2C, kb2=\EOE, kbs=^?,
	kcbt=\E[Z, kcub1=\EOD, kcud1=\EOB, kcuf1=\EOC, kcuu1=\EOA,
	kdch1=\E[3~, kend=\EOF, kent=\EOM, kf1=\EOP, kf10=\E[21~,
	kf11=\E[23~, kf12=\E[24~, kf13=\E[1;2P, kf14=\E[1;2Q,
	kf15=\E[1;2R, kf16=\E[1;2S, kf17=\E[15;2~, kf18=\E[17;2~,
	kf19=\E[18;2~, kf2=\EOQ, kf20=\E[19;2~, kf21=\E[20;2~,
	kf22=\E[21;2~, kf23=\E[23;2~, kf24=\E[24;2~,
	kf25=\E[1;5P, kf26=\E[1;5Q, kf27=\E[1;5R, kf28=\E[1;5S,
	kf29=\E[15;5~, kf3=\EOR, kf30=\E[17;5~, kf31=\E[18;5~,
	kf32=\E[19;5~, kf33=\E[20;5~, kf34=\E[21;5~,
	kf35=\E[23;5~, kf36=\E[24;5~, kf37=\E[1;6P, kf38=\E[1;6Q,
	kf39=\E[1;6R, kf4=\EOS, kf40=\E[1;6S, kf41=\E[15;6~,
	kf42=\E[17;6~, kf43=\E[18;6~, kf44=\E[19;6~,
	kf45=\E[20;6~, kf46=\E[21;6~, kf47=\E[23;6~,
	kf48=\E[24;6~, kf49=\E[1;3P, kf5=\E[15~, kf50=\E[1;3Q,
	kf51=\E[1;3R, kf52=\E[1;3S, kf53=\E[15;3~, kf54=\E[17;3~,
	kf55=\E[18;3~, kf56=\E[19;3~, kf57=\E[20;3~,
	kf58=\E[21;3~, kf59=\E[23;3~, kf6=\E[17~, kf60=\E[24;3~,
	kf61=\E[1;4P, kf62=\E[1;4Q, kf63=\E[1;4R, kf7=\E[18~,
	kf8=\E[19~, kf9=\E[20~, khome=\EOH, kich1=\E[2~,
	kind=\E[1;2B, kmous=\E[M, knp=\E[6~, kpp=\E[5~,
	kri=\E[1;2A, mc0=\E[i, mc4=\E[4i, mc5=\E[5i, meml=\El,
	memu=\Em, oc=\E]104\007, op=\E[39;49m, rc=\E8,
	rep=%p1%c\E[%p2%{1}%-%db, rev=\E[7m, ri=\EM,
	rin=\E[%p1%dT, ritm=\E[23m, rmacs=\E(B, rmam=\E[?7l,
	rmcup=\E[?1049l\E[23;0;0t, rmir=\E[4l, rmkx=\E[?1l\E>,
	rmm=\E[?1034l, rmso=\E[27m, rmul=\E[24m,
	rs1=\Ec\E]104\007, rs2=\E[!p\E[?3;4l\E[4l\E>, sc=\E7,
	setab=\E[%?%p1%{8}%<%t4%p1%d%e%p1%{16}%<%t10%p1%{8}%-%d%e48;5;%p1%d%;m,
	setaf=\E[%?%p1%{8}%<%t3%p1%d%e%p1%{16}%<%t9%p1%{8}%-%d%e38;5;%p1%d%;m,
	setb@, setf@,
	sgr=%?%p9%t\E(0%e\E(B%;\E[0%?%p6%t;1%;%?%p5%t;2%;%?%p2%t;4%;%?%p1%p3%|%t;7%;%?%p4%t;5%;%?%p7%t;8%;m,
	sgr0=\E(B\E[m, sitm=\E[3m, smacs=\E(0, smam=\E[?7h,
	smcup=\E[?1049h\E[22;0;0t, smir=\E[4h, smkx=\E[?1h\E=,
	smm=\E[?1034h, smso=\E[7m, smul=\E[4m, tbc=\E[3g,
	tsl=\E]2;, u6=\E[%i%d;%dR, u7=\E[6n,
	u8=\E[?%[;0123456789]c, u9=\E[c, vpa=\E[%i%p1%dd,
[user@host:~]$
```
I'll have to dig deeper into this as I have no idea what's happening here. I'll probably try a clean install on both desktop and headless server.
Your idea of a separate package is nice but I'd like to keep the installation as simple as possible, at least until I create a proper APT repository for all the packages I currently maintain.

**Comment by barnumbirr** [comment]
Just tried on a fresh Debian 10 install and ran into the same error:

```bash
root@host:~# infocmp alacritty
infocmp: couldn't open terminfo file /usr/share/terminfo/a/alacritty.
root@host:~#
```
Guess I must have done the installation manually on my other device sometime in the past. I'll look into the best way of adding this into the package.

**Comment by barnumbirr** [comment]
Hey @bcbnz,
~~this should now be fixed in [v0.8.0-1](https://github.com/barnumbirr/alacritty-debian/releases/tag/v0.8.0-1).~~ Can you please try it out and let me know how it goes?

**EDIT**: Scratch that, done did a boo-boo. [v0.8.0-3](https://github.com/barnumbirr/alacritty-debian/releases/tag/v0.8.0-3) fixes that.
Cheers.

**Comment by RaZ0rr-Two** [comment]
I have the same issue. But the alacritty files is present in `/usr/share/terminfo/a/alacritty` . But the 

`infocmp alacritty`

shows

`infocmp: couldn't open terminfo file /usr/share/terminfo/a/alacritty.`

**Comment by adongy** [comment]
Hello, and thanks for the package!

I've encountered the same issue. According to the official install docs, the correct command needs to compile the terminfo.

https://github.com/alacritty/alacritty/blob/master/INSTALL.md#terminfo

Here is a patch:

```
diff --git i/debian/rules w/debian/rules
index 2e34745..7ff186e 100755
--- i/debian/rules
+++ w/debian/rules
@@ -18,7 +18,7 @@ override_dh_auto_install:
        install -Dm0644 "extra/completions/_alacritty" "debian/alacritty/usr/share/vendor-completions/_alacritty"
        install -Dm0644 "extra/logo/alacritty-term.svg" "debian/alacritty/usr/share/pixmaps/com.alacritty.Alacritty.svg"
        install -Dm0644 "extra/linux/Alacritty.desktop" "debian/alacritty/usr/share/applications/com.alacritty.Alacritty.desktop"
-       install -Dm0644 "extra/alacritty.info" "debian/alacritty/usr/share/terminfo/a/alacritty"
+       tic -x -e alacritty,alacritty-direct -o "debian/alacritty/usr/share/terminfo" "extra/alacritty.info"
        sed -i 's/Icon=Alacritty/Icon=com.alacritty.Alacritty/g' "debian/alacritty/usr/share/applications/com.alacritty.Alacritty.desktop"

 override_dh_installexamples:
```

I can send a PR if needed.

**Comment by barnumbirr** [comment]
Should be fixed in [v0.8.0-4](https://github.com/barnumbirr/alacritty-debian/releases/tag/v0.8.0-4).
Don't hesitate to reopen this issue if necessary. 💪
